cli/command/environment_test.go

```diff
@@ -0,0 +1,145 @@
+package command
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/quintilesims/layer0/cli/printer"
+	"github.com/quintilesims/layer0/cli/resolver/mock_resolver"
+	"github.com/quintilesims/layer0/client/mock_client"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+)
+
+func TestCreateEnvironment(t *testing.T) {
+	client, _, envCommand, ctrl := initEnvCommandTest(t)
+	defer ctrl.Finish()
+
+	userData := "user_data"
+	file, close := tempFile(t, userData)
+	defer close()
+
+	createEnvRequest := models.CreateEnvironmentRequest{
+		EnvironmentName:  "name",
+		InstanceSize:     "m3.large",
+		MinClusterCount:  2,
+		UserDataTemplate: []byte(userData),
+		OperatingSystem:  "linux",
+		AMIID:            "ami",
+	}
+
+	environment := &models.Environment{}
+	job := NewJob()
```

Comment:
too much magic with `NewJob` IMO. Makes `job_id` come out of nowhere. 

This reads better and is more explicit IMO:
```
client.EXPECT().
    CreateEnvironment(req).
    Return(models.Job{JobID: "jid"}, nil)

client.EXPECT().
   ReadJob("jid").
   Return(models.Job{Result: "eid"}, nil)

client.EXPECT().
   ReadEnvironment("eid").
   Return(models.Environment{}, nil)
```

Reply:
updated to 
```
job := &models.Job{
    JobID:  "job-id",
    Status: job.Completed.String(),
    Result: "entity-id",
}
```
✅